CONTRIBUTING.md

```diff
@@ -78,21 +78,44 @@ See <https://docs.basedpyright.com/> for IDE integration and other info.
 
 ### mypy
 
-Mypy requires a bit more effort to run, since it doesn't play well with
-editable dependencies, and it doesn't seem to recognize NumType's `numpy-stubs`
-if `numpy` is also installed. You can get around that by running:
+Mypy is used for type-checking and specific tests. It requires more effort to run
+correctly because it doesn't handle editable dependencies well and needs isolation
+from the main numpy installation to recognize NumType's numpy-stubs.
+
+#### General Type Checking
+
+To run mypy generally (excluding the main numpy dependency group):
 
 ```shell
 uv run --no-editable --no-group=numpy mypy .
 ```
 
-In rare cases, the cache can get corrupted, which can lead to incorrect
-results. So in case the CI output of mypy differs from your local output, you
-should delete the `.mypy_cache` directory in the `numtype` root.
+#### Running Mypy Tests
+
+The `test/mypy.py` script handles creating an isolated environment (without numpy)
+for running mypy tests:
+
+```bash
+uv run test/mypy.py [OPTIONS]
+```
+
+#### Cache Issues
+
+In rare cases, the cache (`.mypy_cache` in the `numtype` root) can get corrupted,
+leading to incorrect results. If CI output differs from your local output, delete
+this directory.
 
 The documentation can be found at <https://mypy.readthedocs.io/>, and issues can
 be reported at <https://github.com/python/mypy>.
 
+### pytest
+
+Pytest runs runtime tests located in `test/runtime`. It works out-of-the-box:
+
+```bash
+uv run pytest [OPTIONS]
+```
+
```

Comment:
Ah you must've missed the latest mess I made: https://github.com/numpy/numtype/pull/458

So the `test` dir is gone, and the tests are now nested within each individual package in directories called `@test`.